internal/backend/responders.go

```diff
@@ -13,18 +13,15 @@ type responder interface {
 	handle(ctx context.Context, tx *ent.Tx, snap *snapshot) ([]response.Response, error)
 
 	// getMessageID returns the message ID that this responder targets.
-	getMessageID() string
-
-	// setMessageID sets the message ID that the responder targets.
-	setMessageID(string)
+	getMessageID() imap.InternalMessageID
 }
 
 type exists struct {
-	messageID  string
+	messageID  MessageIDPair
```

Comment:
I'm a bit worried that this is a `MessageIDPair` but that the `setMessageID` is gone -- it's possible that we might add a message to the snapshot with the wrong remote ID, due to a race condition, if this responder is popped from the queue, then the remote ID changes, and then we handle the responder.

Could you perhaps just have the internal ID here and query the remote ID from the DB while handling the responder?

Reply:
Done